core/parsigdb/memory.go

```diff
@@ -224,6 +233,24 @@ func getThresholdMatching(typ core.DutyType, sigs []core.ParSignedData, threshol
 	return nil, false, nil
 }
 
+// rootFromParSigDataSet returns the MessageRoot of data only if the duty is supported,
+// and only if data is not a core.Signature.
+func rootFromParSigDataSet(duty core.Duty, data core.ParSignedData) ([32]byte, error) {
+	// known limitation: sync message and contributions might have different hashes,
+	// ignore them for now.
+	// signature duties don't have a MessageRoot implementation at all, by design.
+	if duty.Type == core.DutySyncMessage || duty.Type == core.DutySyncContribution || duty.Type == core.DutySignature {
+		return [32]byte{}, nil
+	}
+
+	root, err := data.MessageRoot()
+	if err != nil {
+		return [32]byte{}, errors.Wrap(err, "cannot hash tree root ParSignedData")
```

Comment:
The error seems a duplicate for line 88.
Maybe here we can simply `return data.MessageRoot()`?